tests/ui/invalid_pyfunction_signatures.stderr

```diff
@@ -16,7 +16,7 @@ error: expected argument from function definition `y` but got argument `x`
 13 | #[pyo3(signature = (x))]
    |                     ^
 
-error: expected one of: `name`, `pass_module`, `signature`, `text_signature`, `crate`
+error: expected one of: `allow_threads`, `name`, `pass_module`, `signature`, `text_signature`, `crate`
```

Comment:
What I would really want to have here is a test that checks that I cannot get `py: Python<'_>` as my argument when using `allow_threads`. This should not be possible to do as the GIL is released, right?

Reply:
(Ideally, it would also have a nice error message explaining why these two contradict each other.)

Reply:
You would have a compilation error, because `Python` is not `Ungil`, but you're right, I will add that case in the UI tests.

Reply:
As expected, there is a compilation error, but the output will not make @davidhewitt happy :sweat_smile:

Reply:
You can add an error message for `Python`, but not really for things like `&PyAny`

Reply:
> You can add an error message for Python, but not really for things like &PyAny

Indeed, but I think we should do what we can. Sorry for putting more work on your table.

Reply:
Not an issue, it will just wait a few hour but I will do the modification :smiley:

Reply:
(Just for coverage, we should also duplicate the compiler error test for plain and async functions.)

Reply:
I've added a check for `Python<'_>` field for both `allow_threads` and async functions. The error raised by `Bound<'_, PyAny>` for these cases is still ugly, but as said earlier, I don't think it's possible to add a pretty error message, and this is already the case for `Python::allow_threads` tests in *ui/not_send.rs* and *ui/not_send2.rs*.